core/state/statedb.libevm_test.go

```diff
@@ -46,26 +46,28 @@ func TestStateDBCommitPropagatesOptions(t *testing.T) {
 			},
 		},
 	)
-	var rec snapTreeRecorder
-	sdb, err := New(types.EmptyRootHash, NewDatabaseWithNodeDB(memdb, triedb), &rec)
+	trieRec, _ := triedb.Backend().(*triedbRecorder)
```

Comment:
```suggestion
	trieRec := triedb.Backend().(*triedbRecorder)
```

If you're not using the `ok` then it can be left out for type assertions and map retrieval.

*However*, please don't do this in production code as it can panic if the concrete type (the implementation of the interface) isn't actually `*triedbRecorder`. I can show you how to avoid this in a follow-up PR but it's not enough to block this one.

Reply:
I worked around the problem, since your suggested change produced a lint error. This closure is simpler anyway.